src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -175,8 +187,20 @@ export const executeQueries = createAsyncThunk<
     );
   }
 
-  // CONDITIONALLY execute tab query (only if queries are different and needed)
-  if (!queriesEqual && needsActiveTabQuery) {
+  // Execute visualization tab query for dynamic tab selection
+  if (needsVisualizationTabQuery) {
+    promises.push(
+      dispatch(
+        executeTabQuery({
```

Comment:
would it be possible if the previous query is still running? is there a way to abort the previous one?

Reply:
Yes, it is possible for a previous query to still be running when this code executes. However, there IS abort handling in place, but it's implemented at the individual query level rather than at the orchestration level. In executeQueryBase, each query creates its own AbortController. 

This is not very accurate. Compared to use_search in discover, since we have multiple query now, the current implementation has a gap where concurrent queries for the same cache key are not properly managed. I will send a follow up PR for this issue.